Jenkinsfile.updateReadme

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2017-present Sonatype, Inc. All rights reserved.
+ * Includes the third-party code listed at http://links.sonatype.com/products/nexus/attributions.
+ * "Sonatype" is a trademark of Sonatype, Inc.
+ */
+@Library('ci-pipeline-library') _
+import com.sonatype.jenkins.pipeline.GitHub
+import com.sonatype.jenkins.pipeline.OsTools
+
+node('ubuntu-zion') {
+  def branch,
+      organization = 'sonatype',
+      dockerHubRepository = 'nexus-iq-server'
+  try {
+    stage('Preparation') {
+      deleteDir()
+      def checkoutDetails = checkout scm
+      branch = checkoutDetails.GIT_BRANCH == 'origin/master' ? 'master' : checkoutDetails.GIT_BRANCH
+    }
+    if (branch == 'master') {
+      stage('Push README.md') {
+        withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'docker-hub-credentials',
+            usernameVariable: 'DOCKERHUB_API_USERNAME', passwordVariable: 'DOCKERHUB_API_PASSWORD']]) {
+          response = OsTools.runSafe(this, """
+            curl -X POST https://hub.docker.com/v2/users/login/ \
+              -H 'cache-control: no-cache' -H 'content-type: application/json' \
+              -d '{ "username": "${env.DOCKERHUB_API_USERNAME}", "password": "${env.DOCKERHUB_API_PASSWORD}" }'
+          """)
+          token = readJSON text: response
+          def dockerHubApiToken = token.token
+
+          def readme = readFile file: 'README.md', encoding: 'UTF-8'
+          readme = readme.replaceAll("(?s)<!--.*?-->", "")
+          readme = readme.replace("\"", "\\\"")
+          readme = readme.replace("\n", "\\n")
+          response = httpRequest customHeaders: [[name: 'authorization', value: "JWT ${dockerHubApiToken}"]],
+              acceptType: 'APPLICATION_JSON', contentType: 'APPLICATION_JSON', httpMode: 'PATCH',
+              requestBody: "{ \"full_description\": \"${readme}\" }",
+              url: "https://hub.docker.com/v2/repositories/${organization}/${dockerHubRepository}/"
+        }
+      }
+    }
+    else {
+      stage('Check README.md') {
+        def readme = readFile file: 'README.md', encoding: 'UTF-8'
+            readme = readme.replaceAll("(?s)<!--.*?-->", "")
+            readme = readme.replace("\"", "\\\"")
+            readme = readme.replace("\n", "\\n")
```

Comment:
The readme could be processed before checking master branch, since you want it unconditionally.

Reply:
> Could the readme be processed before checking master branch, since you want it unconditionally? at least a private method?

That makes more sense, done in https://github.com/sonatype/docker-nexus-iq-server/pull/55/commits/8122b877b52f21522169ec9607eb1730e6e04d77